api/server/publications.ts

```diff
@@ -24,34 +28,33 @@ export function initPublications() {
     return Messages.collection.find({
       chatId
     }, {
-      sort: {createdAt: -1},
+      sort: { createdAt: -1 },
       limit: 30 * messagesBatchCounter
     });
   });
 
-  Meteor.publishComposite('chats', function() {
+  Meteor.publishComposite('chats', function(): PublishCompositeConfig<Chat> {
```

Comment:
You're not right with your thinking for the simple reason that you wanna minimize the amount of data consumed by the client for a fast and optimized networking experience. The smaller the request is, the faster and smoother the application is going to be.